src/shared/components/case-editor/case-event-completion/case-event-completion.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { StateMachine } from '@edium/fsm';
+import { EventCompletionStateMachineService, WorkAllocationService } from '../services';
+import { SessionStorageService } from '../../../services';
+import { EventCompletionComponentEmitter, EventCompletionStateMachineContext } from '../domain';
+import { EventCompletionParams } from '../domain/event-completion-params.model';
+
+@Component({
+  selector: 'ccd-case-event-completion',
+  templateUrl: './case-event-completion.html'
+})
+export class CaseEventCompletionComponent implements OnChanges, EventCompletionComponentEmitter {
+
+  @Input()
+  public eventCompletionParams: EventCompletionParams;
+
+  @Output()
+  eventCanBeCompleted: EventEmitter<boolean> = new EventEmitter();
+
+  public stateMachine: StateMachine;
+  public context: EventCompletionStateMachineContext;
+
+  constructor(private service: EventCompletionStateMachineService,
```

Comment:
Can we make quantify the service with `readonly` again

Reply:
This cannot be made readonly as we are instantiating the object down the line.

Reply:
Thank you @kuda1992, I have now updated the code as you suggested.